### PR TITLE
support cpp lang in `kvproto`

### PR DESCRIPTION
**Pull request by hanfei1991**
1. Support cpp lang.
2. Add protos for learner sync.
3. Add protos for communication between flash and spark.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,7 @@ Cargo.lock
 .vscode
 .idea
 _tools
+
+cpp/kvproto/*.h
+cpp/kvproto/*.cc
```
Comment (zhangjinpeng1987) [inline .gitignore]: Why ignore cpp code?

Comment (hanfei1991) [reply .gitignore]: @flowbehappy I have the same question

Comment (flowbehappy) [reply .gitignore]: Because the cpp code generated by protobuf is likely unportable with other platform.

**Comment by tiancaiamao** [comment]
> Add protos for communication between flash and spark.
Is it necessary to put it in this repo?

**Comment by hanfei1991** [comment]
I've removed the flashpb in this repo.
@zhangjinpeng1987 @tiancaiamao @flowbehappy PTAL again

**Comment by tiancaiamao** [comment]
LGTM

**Comment by BusyJay** [comment]
@overvenus PTAL

**Comment by BusyJay** [comment]
Can someone familiar with the API changes fix the broken code in TiKV side?

**Pull request merged**